lib/tools/system-calls.js

```diff
@@ -855,42 +871,73 @@ systemCallMethods.reboot = async function reboot (retries = DEFAULT_ADB_REBOOT_R
   });
 };
 
+/**
+ * @typedef {Object} rootResult
+ * @property {boolean} isSuccessful True if the call to root/unroot was successful
+ * @property {boolean} wasAlreadyRooted True if the device was already rooted
+ */
+
 /**
  * Switch adb server to root mode.
- *
- * @return {boolean} True of the switch was successful or false
- *                   if the switch failed.
+ * @param {boolean} isElevated - Should we elevate to to root or unroot? (default true)
+ * @return {rootResult}
  */
-systemCallMethods.root = async function root () {
+systemCallMethods.changeUserPrivileges = async function changeUserPrivileges (isElevated) {
+  const cmd = isElevated ? 'root' : 'unroot';
+
+  // If it's already rooted, our job is done. No need to root it again.
+  const isRoot = await this.isRoot();
+  if ((isRoot && isElevated) || (!isRoot && !isElevated)) {
+    return {isSuccessful: true, wasAlreadyRooted: isRoot};
+  }
+
+  let wasAlreadyRooted = false;
   try {
-    let {stdout} = await exec(this.executable.path, ['root']);
+    let {stdout} = await exec(this.executable.path, [cmd]);
 
     // on real devices in some situations we get an error in the stdout
-    if (stdout && stdout.indexOf('adbd cannot run as root') !== -1) {
-      throw new Error(stdout.trim());
+    if (stdout) {
+      if (stdout.includes('adbd cannot run as root')) {
+        log.warn('Cannot run device as root');
```

Comment:
I think we can skip this log at all otherwise it will appear too often on real devices